enhancements/ingress/ingress-fault-detection.md

```diff
@@ -0,0 +1,174 @@
+Often times when networking-related bugs are identified within a cluster, determining which networking sub-component
+(router, SDN, DNS, etc.) is responsible for the observed bug is difficult. Bug reports tend to spend lots of time bouncing between different teams
+as engineers are not sure which component(s) is legitimately at fault. Being able to quickly pinpoint Ingress related issues will expedite the bugfix process.
+
+### Goals
+
+1. Teach the [Ingress Operator](https://github.com/openshift/cluster-ingress-operator) to automatically detect cluster Ingress issues
+and alert cluster administrators when relevant problems are discovered.
+2. Provide a secondary tool that can be triggered at will by a Cluster Administrator or an OpenShift support engineer that spawns an Ingress "debug"
+container that quickly identifies and triages potential Ingress issues within the cluster.
+
+### Non-Goals
+
+1. Teach the Ingress Operator to "self-heal" when cluster Ingress errors or issues are observed.
+2. Modify the [Ingress Controller API](https://github.com/openshift/api/blob/master/operator/v1/types_ingress.go)
+to allow Cluster Administrators to disable the periodic Ingress Operator fault detector at will for a given Ingress Controller.
+
+## Proposal
+
+Once the Ingress Operator has successfully deployed the default [Ingress Controller](https://github.com/openshift/cluster-ingress-operator/blob/master/README.md)
+within a cluster, the Ingress Operator will spawn a simple HTTP echo server via a canary daemonset. Currently in OCP 4.7, this echo server is the trivial
+[Hello OpenShift](https://hub.docker.com/r/openshift/hello-openshift/) image. In future releases, the ingress operator binary will include an echo server sub-command
+(ie `ingress-operator serve-healthcheck`) so that the ingress operator no longer relies on Hello OpenShift.
+The Ingress Operator will also create the necessary Service and Route resources to provide Ingress to the echo container within the `openshift-ingress-canary`
+namespace over HTTP and HTTP/2 connections. Note that a custom certificate will be needed for any HTTP/2 enabled Routes.
+Then, the Ingress Operator will periodically attempt to send a request to the echo container(s), and await a proper response. These requests will be sent
+such that they are routed outside of the cluster to the cloud load balancer first (when applicable), in order to ensure that each part of the Ingress "chain" is being tested.
+If the Ingress Operator is unable to send requests to the echo container(s), the Ingress Operator fails to receive any responses
+from the echo container(s), or the entire request and response flow takes an unusual amount of time, the default Ingress Controller will become degraded as long as sustained failures are observed.
+The ingress canary container(s) will be reachable via a route resource that periodically rotates it's target port. This will cycle the canary endpoints used by the
```

Comment:
```suggestion
The ingress canary container(s) will be reachable via a route resource that periodically rotates its target port. This will cycle the canary endpoints used by the
```